recipes/mxnet/meta.yaml

```diff
@@ -0,0 +1,145 @@
+    folder: 3rdparty/onnx-tensorrt
+  - url: https://github.com/NVlabs/cub/archive/0158fa1.tar.gz
+    sha256: 43424c4c17a997d1d730c89ec14688671245de7941e02b388d7d3df6ea53777a
+    folder: 3rdparty/nvidia_cub
+  - url: https://github.com/nih-at/libzip/releases/download/v1.8.0/libzip-1.8.0.tar.gz
+    sha256: 30ee55868c0a698d3c600492f2bea4eb62c53849bcf696d21af5eb65f3f3839e
+    folder: 3rdparty/libzip
+  - url: https://github.com/kpu/intgemm/archive/8f28282.tar.gz
+    sha256: bc8bd8015613a13747eb769876385ec53e8c1ea7ae3f8414521dc53b8fcdfc65
+    folder: 3rdparty/intgemm
+  - url: https://github.com/oneapi-src/oneDNN/archive/f40443c.tar.gz
+    sha256: da1dd1bb68ce3318db11e57971f975653e84adcb7d65cd879e558f2245e6de9a
+    folder: 3rdparty/onednn
+
+build:
+  number: {{ build_number }}
+  skip: True  # [win or cuda_compiler_version == "10.2"]
+  string: py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}_{{ proc_type }}  # [cuda_compiler_version == "None"]
+  string: py{{ CONDA_PY }}{{ cuda_compiler_version|replace('.', '') }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}_{{ proc_type }}  # [cuda_compiler_version != "None"]
+  run_exports:
+    # ensure matching proc-type
+    - mxnet =*=*_{{ proc_type }}
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - {{ compiler('fortran') }}
+    - {{ compiler('cuda') }}  # [cuda_compiler_version != "None"]
+    - openblas
+    - make
+    - cmake
+    - ninja
+    - ccache
+    - llvm-openmp  # [osx]
+    - libgomp  # [linux]
+    - python
+    - cython
+    - pip
+    - setuptools
+    - libjpeg-turbo
+    - pkgconfig
+    - pkg-config
+
+  host:
+    # need to better expose these deps
+    - libgomp  # [linux]
+    - llvm-openmp  # [osx]
+    - openblas
+    - mkl  # [(target_platform != "osx-arm64") and (cuda_compiler_version == "None")]
+    - mkl-include  # [(target_platform != "osx-arm64") and (cuda_compiler_version == "None")]
+    - opencv
+    - onednn
+    - cudnn  # [cuda_compiler_version != "None"]
+    - nccl  # [cuda_compiler_version != "None"]
+    - numpy >=1.17
+    - requests >=2.20.0,<3
+    - python-graphviz >=0.8.1,<0.9.0
+    - python_version
+    - libjpeg-turbo
+    - python
+
+  run:
+    # ?
+    - {{ pin_compatible('libgomp') }}  # [linux]
+    - {{ pin_compatible('llvm-openmp') }}  # [osx]
+    - openblas
+    - {{ pin_compatible('mkl') }}  # [(target_platform != "osx-arm64") and (cuda_compiler_version == "None")]
+    - {{ pin_compatible('mkl-include') }}  # [(target_platform != "osx-arm64") and (cuda_compiler_version == "None")]
+    - {{ pin_compatible('opencv') }}
+    - {{ pin_compatible('onednn') }}
+    - {{ pin_compatible('cudnn') }}  # [cuda_compiler_version != "None"]
+    - {{ pin_compatible('nccl') }}  # [cuda_compiler_version != "None"]
+    - numpy >=1.17
+    - requests >=2.20.0,<3
+    - python-graphviz >=0.8.1,<0.9.0
+    - python_version
+    - {{ pin_compatible('libjpeg-turbo') }}
```

Comment:
This is going to cause inter-operability problems. You should simply build with jpeg. Do you need turbo? or is it just for performance?

Reply:
I believe they explicitly require turbo, but I can look deeper into it. It is actually _built_ against libjpeg-turbo and I am not really sure if it is a runtime dep. I added all this pinning because when I was testing locally, I would run into problems like "can't find library libopencv.xxx.405" so I thought ooops, I gotta pin them all to start and then relax them later. What's your recommendation here? We potentially will have all sorts of problematic pinning because these are used in the building, not merely as a flexible runtime deps and so this is quite challenging for me to understand as a beginner!